test/AutoDiff/transposing_attr_type_checking.swift

```diff
@@ -0,0 +1,418 @@
+
+  @transposing(Float.threeParams, wrt: (0, 1, 2))
+  func threeParamsT6(s: Float) -> (Float, Double, Float) {
+    return (s + Float(self), Double(s) + self, s + Float(self))
+  }
+
+  @transposing(Float.threeParams, wrt: self) 
+  func threeParamsT6(_ x: Float, _ y: Double, _ z: Float) -> Float {
+    return Float(self + y) + x
+  }
+
+  @transposing(Float.threeParams, wrt: (self, 0)) 
+  func threeParamsT7(_ y: Double, _ z: Float) -> (Float, Float) {
+    let ret = Float(self) + Float(y) + z
+    return (ret, ret)
+  }
+
+  @transposing(Float.threeParams, wrt: (self, 1)) 
+  func threeParamsT7(_ x: Float, _ z: Float) -> (Float, Double) {
+    return (Float(self) + x + z, self + Double(x + z))
+  }
+
+  @transposing(Float.threeParams, wrt: (self, 2)) 
+  func threeParamsT9(_ x: Float, _ y: Double) -> (Float, Float) {
+    let ret = Float(self + y) + x
+    return (ret, ret)
+  }
+
+  @transposing(Float.threeParams, wrt: (self, 0, 1)) 
+  func threeParamsT10(_ z: Float) -> (Float, Float, Double) {
+    let retF = Float(self) + z
+    return (retF, retF, self + Double(z))
+  }
+
+  @transposing(Float.threeParams, wrt: (self, 0, 2)) 
+  func threeParamsT11(_ y: Double) -> (Float, Float, Float) {
+    let ret = Float(self + y)
+    return (ret, ret, ret)
+  }
+
+  @transposing(Float.threeParams, wrt: (self, 0, 1, 2)) 
+  func threeParamsT12() -> (Float, Float, Double, Float) {
+    return (Float(self), Float(self), self, Float(self))
+  }
+}
+
+// Nested struct
+struct level1 {
+  struct level2 {
+    func foo(x: Float) -> Float {
+      return x
+    }
+  }
+}
+
+extension Float {
+  @transposing(level1.level2.foo, wrt: 0)
+  func t(x: level1.level2) -> Float {
+    return self
+  }
+
+  @transposing(level1.level2.foo, wrt: (self, 0))
+  func t() -> (level1.level2, Float) {
+    return (level1.level2(), self)
+  }
+}
+
+// Generics
+extension Float {
+  func genericOneParamFloatOneParam<T: Differentiable & BinaryFloatingPoint>(
+    t: T, f: Float
+  ) -> Float where T == T.TangentVector {
+    return f
+  }
+
+  @transposing(Float.genericOneParamFloatOneParam, wrt: 0)
+  func genericOneParamFloatOneParamT1<T: Differentiable & BinaryFloatingPoint>(
+    f1: Float, f2: Float
+  ) -> T where T == T.TangentVector {
+    return T(1)
+  }
+
+  @transposing(Float.genericOneParamFloatOneParam, wrt: (0, 1))
+  func genericOneParamFloatOneParamT2<T: Differentiable & BinaryFloatingPoint>(
+    f1: Float
+  ) -> (T, Float) where T == T.TangentVector {
+    return (T(1), 1)
+  }
+
+  @transposing(Float.genericOneParamFloatOneParam, wrt: (self, 1))
+  func genericOneParamFloatOneParamT1<T: Differentiable & BinaryFloatingPoint>(
+    t: T
+  ) -> (Float, Float) where T == T.TangentVector {
+    return (1, 1)
+  }
+
+  @transposing(Float.genericOneParamFloatOneParam, wrt: (self, 0, 1))
+  func genericOneParamFloatOneParamT1<
+    T: Differentiable & BinaryFloatingPoint
+  >() -> (Float, T, Float) where T == T.TangentVector {
+    return (1, T(1), 1)
+  }
+}
```

Comment:
Add a trailing line.
```suggestion
}

```